osquery/tables/forensic/carves.cpp

```diff
@@ -116,7 +117,8 @@ QueryData genCarves(QueryContext& context) {
       LOG(WARNING) << "Error inserting new carve entry into the database: "
                    << s.getMessage();
     } else {
-      Dispatcher::addService(std::make_shared<Carver>(paths, guid));
+      auto requestId = Distributed::getCurrentRequestId();
```

Comment:
Probably not an issue not but if we have multiple threads doing distributed queries, we could get two carves with the same requestID I think.